app/core/components/Comments.tsx

```diff
@@ -0,0 +1,208 @@
+  useEffect(() => {
+    if (transition.type == "actionRedirect") {
+      setOpenDeleteModal(false);
+      setOpenEditComment(false);
+    }
+  }, [transition]);
+
+  return (
+    <Paper sx={{ padding: 2, marginY: 2 }}>
+      <Grid container spacing={2} alignItems="center">
+        <Grid item>
+          <Avatar alt={"alt"} src={comment.author?.avatarUrl ?? ""}></Avatar>
+        </Grid>
+        <Grid justifyContent="left" item xs zeroMinWidth>
+          <Typography variant="body1">{`${comment.author?.firstName} ${comment.author?.lastName}`}</Typography>
+          <Typography variant="body2">
+            {comment.updatedAt.toLocaleString()}
+          </Typography>
+        </Grid>
+        {profileId === comment.authorId && (
+          <Grid item justifyContent="right">
+            <IconButton
+              color="warning"
+              aria-label="delete"
+              onClick={() => setOpenDeleteModal(true)}
+            >
+              <Delete />
+            </IconButton>
+            <IconButton
+              aria-label="edit"
+              color="primary"
+              onClick={() => setOpenEditComment(true)}
+            >
+              <Edit />
+            </IconButton>
+          </Grid>
+        )}
+        <Grid item xs={12}>
+          <Box>
+            <Markdown>{comment.body}</Markdown>
+          </Box>
+          {comment.children &&
+            comment.children.map((commentChild) => {
+              return (
+                <CommentItem
+                  key={commentChild.id}
+                  comment={commentChild}
+                  projectId={projectId}
+                  profileId={profileId}
+                />
+              );
+            })}
+          {!comment.parentId && (
+            <ValidatedForm
+              action={`./comment/create`}
+              method="post"
+              validator={validator}
+              resetAfterSubmit
+            >
+              <TextEditor name="body" height={100} />
+              <input type="hidden" name="parentId" value={comment.id} />
+              <input type="hidden" name="projectId" value={projectId} />
+              <br />
+              <Button type="submit" variant="contained">
+                Reply
+              </Button>
+            </ValidatedForm>
+          )}
+        </Grid>
+      </Grid>
+      <ModalBox open={openDeleteModal} close={() => setOpenDeleteModal(false)}>
+        <h2>Are you sure you want to delete this comment?</h2>
+        <p>This action cannot be undone.</p>
+        <Form action={`./comment/${comment.id}`} method="delete">
+          <Button variant="contained" onClick={() => setOpenDeleteModal(false)}>
+            Cancel
+          </Button>
+          &nbsp;
+          <Button type="submit" variant="contained" color="warning">
+            Delete
+          </Button>
+        </Form>
+      </ModalBox>
+      <ModalBox open={openEditComment} close={() => setOpenEditComment(false)}>
```

Comment:
In personal I think that [Dialog](https://mui.com/material-ui/react-dialog/) is better than modal, because already have a structure for the items inside it. Maybe we can consider change the modal with dialog, but it's a personal opinion 😄

Reply:
Yes! I saw you using dialogs on the other PR. In my case I briefly tried them, but using the ModalBox I didn't need to add any styling or theming, and things looked good. But when I tried the Dialog, the padding was off, so I switched back to the ModalBox instead. I agree that Dialog seems to be the better option on paper, though, since the docs also say that 😅